gravitee-apim-console-webui/src/management/api/notifications/notifications-list/notofication-details/notification-details.component.ts

```diff
@@ -13,11 +13,107 @@
+import { Component, Inject, OnDestroy, OnInit } from '@angular/core';
+import { takeUntil, tap } from 'rxjs/operators';
+import { combineLatest, Subject } from 'rxjs';
+import { groupBy, keysIn } from 'lodash';
+import { FormControl, FormGroup } from '@angular/forms';
+
+import { UIRouterStateParams } from '../../../../../ajs-upgraded-providers';
+import { NotificationSettingsService } from '../../../../../services-ngx/notification-settings.service';
+import { SnackBarService } from '../../../../../services-ngx/snack-bar.service';
+import { NotificationSettings } from '../../../../../entities/notification/notificationSettings';
+import { Notifier } from '../../../../../entities/notification/notifier';
+import { Hooks } from '../../../../../entities/notification/hooks';
+
+type HooksForm = {
+  list: any;
+  categories: string[];
+};
 
 @Component({
   selector: 'notifications-details',
   template: require('./notification-details.component.html'),
   styles: [require('./notification-details.component.scss')],
 })
-export class NotificationDetailsComponent {}
+export class NotificationDetailsComponent implements OnInit, OnDestroy {
+  public isLoadingData = true;
+  public hooks: HooksForm;
+  public notificationSettings: NotificationSettings;
+  public formInitialValues: unknown;
+  public notifier: Notifier;
+  notificationForm: FormGroup;
+
+  private unsubscribe$: Subject<boolean> = new Subject<boolean>();
+
+  constructor(
+    private readonly notificationSettingsService: NotificationSettingsService,
+    private readonly snackBarService: SnackBarService,
+    @Inject(UIRouterStateParams) private readonly ajsStateParams,
+  ) {}
+
+  public ngOnInit() {
+    this.isLoadingData = true;
+    combineLatest([
+      this.notificationSettingsService.getHooks(),
+      this.notificationSettingsService.getSingleNotificationSetting(this.ajsStateParams.apiId, this.ajsStateParams.notificationId),
+      this.notificationSettingsService.getNotifiers(this.ajsStateParams.apiId),
+    ])
+      .pipe(
+        tap(([hooks, notificationSettings, notifiers]) => {
+          this.hooks = {
+            list: hooks.map((categories: Hooks) => ({
+              ...categories,
+              checked: notificationSettings.hooks.includes(categories.id),
+            })),
+            categories: keysIn(groupBy(hooks, 'category')),
+          };
+          this.notificationSettings = notificationSettings;
+          this.notifier = notifiers.find((i) => i.id === this.notificationSettings.notifier);
+
+          this.notificationForm = new FormGroup({
+            notifier: new FormControl(this.notificationSettings.config),
+          });
+
+          this.hooks.list.map((item) => {
+            this.notificationForm.addControl(`${item.id}`, new FormControl(item.checked));
+          });
+
+          this.formInitialValues = this.notificationForm.getRawValue();
+
+          this.hooks.list = groupBy(this.hooks.list, 'category');
```

Comment:
What do you think about making this function a little clearer with hooks?

For example, :
1. create const to list hooks and map to found checked one. 
2. use it to init notificationFormGroup 
3. group it by category
4. _.map to the group to convert to CategoriesHooksVM and save it in this.***

Reply:
Done